ide-support/revsblibrary.livecodescript

```diff
@@ -702,6 +702,101 @@ function revSBTargetToPlatform pTarget
    end if
 end revSBTargetToPlatform
 
+-- item 1 of the architectures should be the default architecture
+constant kWindowsArchitectures = "x86-32"
+constant kMacOSXArchitectures = "x86-32,x86-64"
+constant kLinuxArchitectures =  "x86-32,x86-64,armv6-hf"
+constant kAndroidArchitectures =  "armv6"
+constant kiOSArchitectures =  "armv7,arm64"
+constant kEmscriptenArchitectures =  "js"
+
+function revSBPlatformArchitectures pPlatform
+   switch pPlatform
+      case "Windows"
+         return kWindowsArchitectures
+      case "MacOSX"
+         return kMacOSXArchitectures
+      case "Linux"
+         return kLinuxArchitectures
+      case "Android"
+         return kAndroidArchitectures
+      case "iOS"
+         return kiOSArchitectures
+      case "Emscripten"
+         return kEmscriptenArchitectures
+      default
+         return empty
+   end switch
+end revSBPlatformArchitectures
+
+private function __ArchitectureSynonyms pArchitecture
+   switch pArchitecture
+      case "x86-32"
+         return pArchitecture,"x86,i386"
+      case "x86-64"
+         return pArchitecture,"x64"
+   end switch
+   
+   return pArchitecture
+end __ArchitectureSynonyms
+
+private command __ExternalArchitecture pPlatform, @xName, @rArchitecture
+   local tArchitectures
+   put revSBPlatformArchitectures(pPlatform) into tArchitectures
+   
+   local tArchitecture,tFoundArchitecture
+   repeat for each item tArchitecture in tArchitectures
+      local tArchitectureIdentifier
+      repeat for each item tArchitectureIdentifier in __ArchitectureSynonyms(tArchitecture)
+         if xName ends with tArchitectureIdentifier then
+            put tArchitecture into tFoundArchitecture 
+            set the itemDelimiter to "-"
+            if the number of items of xName > 1 then
+               delete item -(the number of items of tArchitectureIdentifier) to -1 of xName
+            else
+               set the itemDelimiter to "_"
+               if the number of items of xName > 1 then
+                  delete item -(the number of items of tArchitectureIdentifier) to -1 of xName
+               end if
+            end if
+            
+            exit repeat
+         end if
+         set the itemDelimiter to comma
+      end repeat
+      
+      if tFoundArchitecture is not empty then
+         exit repeat
+      end if
+   end repeat
+   
+   if tFoundArchitecture is empty then
+      put item 1 of tArchitectures into tFoundArchitecture
```

Comment:
The itemdelimiter won't be comma here if the exit repeat on line 763 is hit, will it? I think it needs to be reset at some point.